### ISSUE TITLE
Can't run basic aac standalone example

### ISSUE THREAD

**Issue by swordsreversed** [StreamMachine/StreamMachine#94]
If i use mp3 with the basic standalone example in the README it works, but if i try aac/mp4 I cannot get it to work. Am i missing something basic?

```
 ./streammachine-cmd --config config/standalone.json --title "StreamMachine"

./streammachine-util-cmd source --port 8002 --stream curry --password three ./dustaac.mp4
```

My config:

```
{
	"port":         8001,
	"source_port":  8002,
        "api_port":     8003,
     "log": {
      "stdout":     true,
      "json": {
        "file": "/var/log/streammachine/streammachine.log",
        "level": "info"
      }
  },
  "streams": {
    "curry": {
      "key":              "curry",
      "root_route":       true,
      "format":           "aac",
      "seconds":          3600,
      "metaTitle":        "curry",
      "host":             "localhost",
      "source_password":  "three"
    }
  }
}
```

Errors i'm getting: 

```
5520/slave -- debug: In slave configureStreams with  key=curry, root_route=true, format=aac, seconds=3600, metaTitle=curry, host=localhost, source_password=three, segment_duration=10, limit_full_index=false, preroll=, transcoder=, log_interval=30000, meta_interval=32768, max_buffer=4194304, burst=30, fallback=null, acceptSourceMeta=false, log_minutes=true, monitored=false, metaUrl=, preroll_key=, group=null, bandwidth=0, codec=null, ffmpeg_args=null, stream_key=null, impression_delay=5000
5520/slave -- debug: Starting up stream: curry key=curry, root_route=true, format=aac, seconds=3600, metaTitle=curry, host=localhost, source_password=three, segment_duration=10, limit_full_index=false, preroll=, transcoder=, log_interval=30000, meta_interval=32768, max_buffer=4194304, burst=30, fallback=null, acceptSourceMeta=false, log_minutes=true, monitored=false, metaUrl=, preroll_key=, group=null, bandwidth=0, codec=null, ffmpeg_args=null, stream_key=null, impression_delay=5000
5520/slave -- debug: Enabling HLS Index for stream. stream=curry
5520 -- debug: mapping master -> slave on curry
5520/slave -- debug: Slave stream got source connection stream=curry
5520/slave -- debug: RewindBuffer got source event stream=curry
5520/slave -- debug: Preroll settings are  preroll=, stream=curry
5520/slave -- debug: Stream's max buffer size is 4194304 stream=curry
5520/slave -- debug: Rewind's max buffer length is  max=0, seconds=3600, stream=curry
5520/slave -- info: RewindBuffer is now at  seconds=NaN, length=0, stream=curry
5520/slave -- debug: Slave source loaded rewind buffer. stream=curry
5520/sourcein -- debug: Incoming source attempt.
5520/sourcein -- debug: ICY SOURCE attempt. url=/curry
5520/sourcein -- debug: Trying to authenticate ICY source for curry
5520/sourcein -- debug: ICY source authenticated for curry.
5520/master -- debug: New Icecast source. source_mount=curry, uuid=15f92f43-c9d5-4c5c-affa-cc30f1ba08ac
5520/master -- event: Promoting new source to active. source=Icecast (::ffff:127.0.0.1:37550), source_mount=curry
5520/master -- error: Failed to get source vitals before timeout. Forcing disconnect. source_mount=curry, uuid=15f92f43-c9d5-4c5c-affa-cc30f1ba08ac
5520/master -- debug: Setting _isDisconnected source_mount=curry, uuid=15f92f43-c9d5-4c5c-affa-cc30f1ba08ac
5520/master -- event: Inactive source disconnected. source_mount=curry
5520/master -- error: useSource failed to get switchover within five seconds. new_source=Icecast (::ffff:127.0.0.1:37550), old_source=undefined, source_mount=curry
```

Please help!

**Comment by swordsreversed** [comment]
I subsequently have tried boradcasting from my mac using BUTT. Once again mp3 is working fine but aac connects then gives the error:

```
PUSH IN MIDDLE NOT IMPLEMENTED 1477023624876 1477023616239 1477023638255
```

Is there something missing on my ubuntu 14 server @ewr? I've compiled ffmpeg with libfdkaac. Is there a library I need?

Would really love to get this sotware working. Thanks in advance.

**Comment by swordsreversed** [comment]
Ok for the big dummies like me trying to use AAC as an input source, the simple answer is DON'T. Although I can't find explicit reference to it, it is seemingly impossible due to the nature of the format's structure. Simply send mp3 instead and use StreamMachine to encode an AAC stream for you.
